### ISSUE TITLE
propagator is calling rhs_generate() excessively

### ISSUE THREAD

**Issue by vdrhtc** [qutip/qutip#555]
In file [propagator.py](https://github.com/qutip/qutip/blob/master/qutip/propagator.py), near line 119:
```python
td_type = _td_format_check(H, c_op_list, solver='me')[2]
if td_type > 0:
    rhs_generate(H, c_op_list, args=args, options=options)
```
The performance would be improved if this code checked for the presence of an already compiled tdfunc like this:
```python
if td_type > 0 and config.tdfunc is None:
    ...
```
which only requires an additional import of the config object from qutip.solver

**Comment by nonhermitian** [comment]
Thanks for pointing this out. Luckily, I already came across this when updating the propagator. Will submit a pull in the next week.

**Comment by nonhermitian** [comment]
This should be fixed in Pull #557.  I just removed the entire `rhs_generate` call in propagator as both sesolve and mesolve handle the building internally.

**Comment by nonhermitian** [comment]
This has been merged.
